include/rdma/fi_domain.h

```diff
@@ -121,6 +126,7 @@ struct fi_mr_attr {
 	void			*context;
 	size_t			auth_key_size;
 	uint8_t			*auth_key;
+	enum fi_hmem_iface	iface;
```

Comment:
A naming nit, but this could be just a hmem type field instead of an interface.. The same type of memory could be managed with different interfaces/libraries.

Reply:
The intention here is to specify the interface/library to use.

Reply:
A provider, or the eventual peer s/w interface implementation, could choose to use different libraries to deal with the same buffer. I think this attribute itself should not determine what library should be used.

Reply:
There's no requirement on the provider implementation.  It's conveying what interfaces the caller is using the manage the memory.  The virtual memory could migrate between different physical locations, moving from host to device, based on access.